cli/internal/run/dry_run.go

```diff
@@ -44,7 +44,11 @@ func DryRun(
 		}
 
 		if taskSummary.Framework == "" {
-			taskSummary.Framework = runsummary.MissingFrameworkLabel
+			if rs.Opts.runOpts.FrameworkInference {
+				taskSummary.Framework = runsummary.NoFrameworkDetected
+			} else {
+				taskSummary.Framework = runsummary.FrameworkDetectionSkipped
```

Comment:
The only reason `MissingFrameworkLabel` is _only_ in dry run, is because it was already there and I didn't want to put it in Real Run summaries. I can't remember why I made that decision though. Maybe I thought that `framework: ""` was just as clear as a placeholder, so I didn't want a placeholder in a new place?

For *skipped* inference, maybe we _do_ want it to say "skipped" even in real run summaries, to differentiate from `framework: ""`?

Or you could make the case that since we have the `frameworkInference: false` at the top level of the summary, you could... uhhh... *infer* the meaning of `framework: ""` in each task? 

I guess I don't have a strong enough opinion, because while leaving this comment, I'm leaning towards adding the Missing _and_ skipped labels into real run summaries, whereas I explicitly decided _not_ to add it before. What are your thoughts?

Reply:
There are two ways to derive this information:
- `.tasks[].framework`
- `.frameworkInference`

And the consequences of the configuration show up in run summary as:
- `.environmentVariables.inferred`

We don't _need_ it; it's derivable, and having an arbitrary string slug as part of an API feels bad. I added this to the PR _primarily_ because I didn't want people getting `NoFrameworkDetected` when it was actually skipped.

The other legit option is to _remove_ it from dry run, but that's an API change and should be held to at least 1.10.

Reply:
(In other words: rather than adding it to run summary, I'd pitch removing it from dry run.)

Reply:
In a post-Go world we could do `Option<Framework>` which would also easily serialize to `null` or `slug`, though since both `""` and `null` are falsey it's not a big difference. (though it may communicate more-clearly at a glance)

Reply:
You can `omitempty` to differentiate too. 

I'm ok with the direction of removing from dry run, and in that case it's fine to only add this differentiation there.

Reply:
I want the field to always be present though! lots of things blow up on missing keys.